File: works/tmp/16968.py
str = input()
sum = 0
# 파이썬의 삼항연산자
sum = 10 if str[0] == 'd' else 26

for i in range(1, len(str)):
    if str[i] == str[i-1]:
        if str[i] == 'd': sum = sum * 9
        else: sum = sum * 25
    else:
        if str[i] == 'd': sum = sum * 10
        else: sum = sum * 26

print(sum)

File: works/tmp/16968.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int main()
{
    string str; cin >> str;
    int sum;
    str[0] == 'd' ? sum = 10 : sum = 26;

    for(int i = 1; i < str.size(); i++)
    {
        // 이전과 같은 문자
        if(str[i] == str[i-1])
        {
            if(str[i] == 'd') sum *= 9;
            else sum *= 25;
        }
        else
        {
            if(str[i] == 'd') sum *= 10;
            else sum *= 26;
        }
    }

    cout << sum;
}